admin/server/usergroups.go

```diff
@@ -0,0 +1,696 @@
+	}
+
+	if orgRole != nil {
+		if !claims.OrganizationPermissions(ctx, org.ID).ReadOrgMembers {
+			return nil, status.Error(codes.PermissionDenied, "not allowed to list usergroup members")
+		}
+	}
+
+	projectRoles, err := s.admin.DB.FindProjectMemberUsergroupRoles(ctx, group.ID, org.ID)
+	if err != nil && !errors.Is(err, database.ErrNotFound) {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	for _, role := range projectRoles {
+		if !claims.ProjectPermissions(ctx, org.ID, role.ProjectID).ReadProjectMembers {
+			return nil, status.Error(codes.PermissionDenied, "not allowed to list usergroup members")
+		}
+	}
+
+	members, err := s.admin.DB.FindUsergroupMembersUsers(ctx, group.ID)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	usergroupMembers := make([]*adminv1.MemberUser, len(members))
+	for i, member := range members {
+		usergroupMembers[i] = &adminv1.MemberUser{
+			UserId:    member.ID,
+			UserEmail: member.Email,
+			UserName:  member.DisplayName,
+		}
+	}
+
+	return &adminv1.ListUsergroupMemberUsersResponse{Members: usergroupMembers}, nil
+}
+
+func (s *Server) RemoveUsergroupMemberUser(ctx context.Context, req *adminv1.RemoveUsergroupMemberUserRequest) (*adminv1.RemoveUsergroupMemberUserResponse, error) {
+	observability.AddRequestAttributes(ctx,
+		attribute.String("args.org", req.Organization),
+		attribute.String("args.usergroup", req.Usergroup),
+	)
+
+	org, err := s.admin.DB.FindOrganizationByName(ctx, req.Organization)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	group, err := s.admin.DB.FindUsergroupByName(ctx, req.Usergroup, org.ID)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	if org.AllUsergroupID != nil && group.ID == *org.AllUsergroupID {
+		return nil, status.Error(codes.InvalidArgument, "cannot remove member from all-users group")
+	}
+
+	user, err := s.admin.DB.FindUserByEmail(ctx, req.Email)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	claims := auth.GetClaims(ctx)
+
+	orgRole, err := s.admin.DB.FindOrganizationMemberUsergroupRole(ctx, group.ID, org.ID)
+	if err != nil && !errors.Is(err, database.ErrNotFound) {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	if orgRole != nil {
+		if !claims.OrganizationPermissions(ctx, org.ID).ManageOrgMembers {
+			return nil, status.Error(codes.PermissionDenied, "not allowed to remove usergroup members")
+		}
+	}
+
+	projectRoles, err := s.admin.DB.FindProjectMemberUsergroupRoles(ctx, group.ID, org.ID)
+	if err != nil && !errors.Is(err, database.ErrNotFound) {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	for _, role := range projectRoles {
+		if !claims.ProjectPermissions(ctx, org.ID, role.ProjectID).ManageProjectMembers {
+			return nil, status.Error(codes.PermissionDenied, "not allowed to remove usergroup members")
+		}
+	}
```

Comment:
Similarly, removing a user should just require `ReadOrgMembers` on the group's org